app/src/main/java/com/lagradost/cloudstream3/extractors/Sendvid.kt

```diff
@@ -0,0 +1,32 @@
+package com.lagradost.cloudstream3.extractors
+
+import com.lagradost.cloudstream3.utils.*
+import com.lagradost.cloudstream3.app
+import com.lagradost.cloudstream3.utils.M3u8Helper.Companion.generateM3u8
+
+class SendvidHttps: Sendvid() {
+    override val mainUrl: String = "https://www.sendvid.com"
+}
+
+open class Sendvid : ExtractorApi() {
+    override var name = "Sendvid"
+    override val mainUrl = "https://sendvid.com"
+    override val requiresReferer = false
+
+    override suspend fun getUrl(url: String, referer: String?): List<ExtractorLink>? {
+        val doc = app.get(url).document
+        val urlString = doc.select("head meta[property=og:video:secure_url]").attr("content")
+        val sources = mutableListOf<ExtractorLink>()
+        if (urlString.contains("m3u8"))  {
+                generateM3u8(
+                    name,
+                    urlString,
+                    mainUrl,
+                    headers = app.get(url).headers.toMap()
+                ).forEach {link ->
+                    sources.add(link)
```

Comment:
Adding sources is what we've done a lot in our extractors, but that is kinda deprecated. It would be much cleaner to use the new https://github.com/recloudstream/cloudstream/blob/master/app/src/main/java/com/lagradost/cloudstream3/utils/ExtractorApi.kt#L456-L463
and then just do `.forEach(callback)`

Reply:
^ not a required change